lib/rdkafka/consumer/message.rb

```diff
@@ -28,6 +28,10 @@ class Message
       # @return [Time, nil]
       attr_reader :timestamp
 
+      # This message's raw timestamp, if provided by the broker
+      # @return [Time, nil]
+      attr_reader :raw_timestamp
```

Comment:
I have a mixed feeling about exposing this. How often does the messages timestamp lookup occurs? That is: does frequency of re-computing the raw time value justifies expanding the API.

Reply:
I'm happy to remove this change from the PR but it might be worth considering the use case of wanting access to the original data as-is. As a wrapper library it makes sense to expose the raw data as much as possible.

Reply:
@timflapper I don't have a super strong opinion. I just don't like expanding APIs unless needed because then you create a long-lasting contract with the users ;)

@thijsc should probably make a decision here

Reply:
I think it would be more Ruby-esque to use times in the input topic partition list. We're trying to make this as nice to use as possible. We could add a field for this on the Ruby TPL and convert it when calling librdkafka.

Reply:
Ok. I'll see if I can get it to work to allow the TPL to detect offset as a Time object to convert to a millisecond timestamp.

My personal note on this is that I would personally choose to keep timestamps in their original form (perhaps naming them timestamp_ms to notify the timestamp format) and letting the library's user convert it to a Time object if they wish. The time conversion means that the library can never be as fast as it can possibly be since every message has a data transformation that may or may not be useful to the user but that does slow down the message processing, even if very slightly.

Reply:
@timflapper this would change the contract for this package. It could however be lazy evaluated for sure. Nonetheless IMHO worth opening a discussion about it outside of scope of this task.

Reply:
@timflapper We made a design choice to have this library get as close to idiomatic Ruby as we can. That means that we sometimes sacrifice some performance.

Reply:
I guess you could do the conversion in a lazy way btw? Often it won't be needed at all.

Reply:
@thijsc you mean introducing a lazy convertion to the message object? Sounds like a code smell. Imho that would be an unexpected mutability of a simple data object (so far at this point).

Reply:
Why not? We we can store the raw number in a private field and convert it whenever we need. Why is that a code smell?

Reply:
For me, the message then is no longer just a "close to being frozen" data structure, and it becomes mutable. From the API standpoint, this approach is **nothing** wrong.